lib/services/local/users.go

```diff
@@ -766,6 +774,48 @@ func sessionDataKey(user, sessionID string) []byte {
 	return backend.Key(webPrefix, usersPrefix, user, webauthnSessionData, sessionID)
 }
 
+// globalSessionDataLimiter keeps a count of in-flight session data challenges
+// over a period of time.
+type globalSessionDataLimiter struct {
+	// Clock is public so it may be overwritten by tests.
+	Clock clockwork.Clock
+	// ResetPeriod is public so it may be overwritten by tests.
+	ResetPeriod time.Duration
+	// mu guards the fields below it.
+	mu         sync.Mutex
+	scopeCount map[string]int
+	lastReset  time.Time
+}
+
+func (l *globalSessionDataLimiter) add(scope string, n int) int {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	// Reset counters to account for key expiration.
+	now := l.Clock.Now()
+	if now.Sub(l.lastReset) >= l.ResetPeriod {
```

Comment:
```suggestion
	if l.Clock.Since(l.lastReset) >= l.ResetPeriod {
```

Does this work? It does with the standard `time` package but I'm not sure if clockwork supports it.

Reply:
It does work, but I use the `now` variable below so keeping it.